clients/src/main/java/org/apache/kafka/common/record/MemoryRecordsBuilder.java

```diff
@@ -376,7 +390,7 @@ public boolean isClosed() {
     }
 
     public boolean isFull() {
-        return isClosed() || this.writeLimit <= estimatedBytesWritten();
+        return isClosed() || (this.numRecords > 0 && this.writeLimit <= estimatedBytesWritten());
```

Comment:
I assume that `writeLimit` is based of batch size? 

So, in cases where `writeLimit == 0`, even an empty batch would be considered full, triggering the NPE on send? 

Is that correct?